pkg/metrics/prometheus_exporter.go

```diff
@@ -28,42 +24,18 @@ func newPrometheusExporter() (view.Exporter, error) {
 	errCh := make(chan error)
 	log.Info("Starting server for OpenCensus Prometheus exporter")
 	// Start the server for Prometheus scraping
-	go func() {
-		srv := startNewPromSrv(e, *prometheusPort)
-		errCh <- srv.ListenAndServe()
-	}()
-	select {
-	case err := <-errCh:
-		if err != nil {
-			return e, err
-		}
-	}
-	return e, nil
-}
-
-func getCurPromSrv() *http.Server {
-	curPromSrvMux.Lock()
-	defer curPromSrvMux.Unlock()
-	return curPromSrv
-}
-
-func resetCurPromSrv() {
-	curPromSrvMux.Lock()
-	defer curPromSrvMux.Unlock()
-	if curPromSrv != nil {
-		curPromSrv.Close()
-		curPromSrv = nil
+	srv := startNewPromSrv(e, *prometheusPort)
+	errCh <- srv.ListenAndServe()
+	err = <-errCh
+	if err != nil {
+		return nil, err
 	}
+	return nil, nil
```

Comment:
Now that view.Exporter is always nil, we should change the function signature to just return error.